Jenkinsfile

```diff
@@ -8,12 +8,41 @@ node {
       checkout scm
     }
 
+    env.AWS_DEFAULT_REGION = 'us-east-1'
+
     stage('cibuild') {
       wrap([$class: 'AnsiColorBuildWrapper']) {
         sh './scripts/cibuild'
       }
     }
 
+    if (env.BRANCH_NAME == 'master' || env.BRANCH_NAME.startsWith('test/') || env.BRANCH_NAME.startsWith('release/') || env.BRANCH_NAME == 'PR-96') {
+      // Publish container images built and tested during `cibuild`
+      // to the private Amazon Container Registry tagged with the
+      // first seven characters of the revision SHA.
+      stage('cipublish') {
+        // Decode the ECR endpoint stored within Jenkins.
+        withCredentials([
+          [
+            $class: 'StringBinding',
+            credentialsId: 'NOAA_AWS_ECR_ENDPOINT',
+            variable: 'NOAA_AWS_ECR_ENDPOINT'
+          ],
+          [
+            $class: 'StringBinding',
+            credentialsId: 'NOAA_AWS_DEPLOY_ROLE_ARN',
+            variable: 'NOAA_AWS_DEPLOY_ROLE_ARN'
+          ],
```

Comment:
I set up a cross-account role to allow the RF Jenkins instance to deploy into the NOAA account.

Reply:
This cross-account role gives me a little bit of anxiety. I've had a hard time wrapping my head around the dependency web every time we've made use of it, and I'm worried we're going to end up with more of these roles in the future.

Is there anyway we could work around this? On our local CI boxes, we create an AWS profile for the various accounts we need to connect to. While not perfect, I think it's easier to follow than the cross-account role. Did you consider something like that? Is there another place we could run CI?

Reply:
Yeah, this is totally just a product of running it on the RF Jenkins instance! I talked about this with Hector during a standup and we decided on the RF instance based purely on team composition, since this doesn't have any overlap with CA or GA. I don't think we were thinking about the extra complexity we'd be bringing on with cross-account roles though.

I see the tradeoff as extra auth complexity vs. picking a team-oriented Jenkins instance as a default dumping ground for anything that isn't cleanly CA, GA, or RF related. I kinda lean towards the former, but I may have gone a little noseblind to that dependency web. What're your thoughts?

Reply:
I think the RF instance makes sense. I'm more concerned about the cross-account layer. As an example, I'm pretty sure there is stuff that isn't codified in this PR that has to be in place to finalize the connection between the RF account and the NOAA account. You just have to know that stuff exists and model it in your head when dealing with this stuff.

An AWS profile would be clear: there's an IAM user that was created, and the RF Jenkins instance consumes it like all our local CI boxes.

I know that we've had success using this with Robarts, but Robarts was a situation where we were doing cross-account stuff within AWS accounts for the same project, so things felt more contained.

Reply:
@rbreslow and I discussed this during standup. We decided to keep role-based authentication in place, but to add a comment indicating that it requires a principal to be added to the RF Jenkins instance profile. We're also going to rename the `aws_azavea_account_id` to `aws_raster_foundry_account_id` to make the trust relationship more obvious.